sharness/t0050-service-state-upgrade-from-current.sh

```diff
@@ -0,0 +1,24 @@
+#! /bin/sh
+
+test_description="Test service state 'upgrade' from current version"
+
+. lib/test-lib.sh
+
+test_ipfs_init
+cleanup test_clean_ipfs
+test_cluster_init
+cleanup test_clean_cluster
+
+test_expect_success IPFS,CLUSTER "cluster-service state preserved by migration" '
+    cid=`docker exec ipfs sh -c "echo testing | ipfs add -q"` &&
+    ipfs-cluster-ctl pin add "$cid" && sleep 2 &&
+    kill -1 $CLUSTER_D_PID && sleep 30 &&
+    ipfs-cluster-service --config "test-config" state upgrade &&
+    ipfs-cluster-service --config "test-config" >"$IPFS_OUTPUT" 2>&1 &
+    sleep 50 &&
+    ipfs-cluster-ctl pin ls "$cid" | grep -q "$cid" &&
+    ipfs-cluster-ctl status "$cid" | grep -q -i "PINNED" &&
+    pkill -1 -f "ipfs-cluster-service --config test-config" && sleep 2
```

Comment:
I have the feeling we should add a kill() start() functions to the lib so we can do this a bit more cleanly.

Reply:
Hmm yeah I agree with you.  I tried adapting the current clean_cluster lib function to take in a pid for this test, but I ran back into the problem that within sharness executable strings `$!` doesn't actually record the latest spawned process.  So `pkill` seems like the best I can do right now.  I am cool with adding a `start` and `kill` that uses `pkill`.

Reply:
start() can save the pid and kill can re-use that, if both are in the lib/test_lib.sh file. it works like that now (it kills on clean up)